sampleExtension/background.js

```diff
@@ -1,60 +1,117 @@
+        blockedSites = blacklist;
+        return callback();
+    });
+};
+
+//Loads the intercept time+date list from storage
+retrieveInterceptDateList = function() {
+    getInterceptDateList(function(dateList) {
+        interceptDateList = dateList;
+    });
+};
+
+//Loads the intercept counter from storage
+retrieveInterceptCounter = function(callback) {
+    getInterceptCounter(function(counter) {
+        interceptCounter = counter;
+        return callback();
     });
 };
 
 // This function adds one url to the blacklist
 addToBlockedSites = function(urlToAdd) {
-    var formatUrl = urlToAdd.split(['//'])[1];
-    formatUrl = '*://' + formatUrl + '*';
-    blockedSites.push(formatUrl);
-    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-    });
+    var blockedSiteItem = new BlockedSite(formatUrl(urlToAdd));
+    blockedSites.push(blockedSiteItem);
+    updateStorage();
+    replaceListener();
+};
+
+// This function adds the current time+date to the saved time+date list
+addToInterceptDateList = function() {
+    var newDate = new Date().toDateString();
+    if (interceptDateList == null) {
+        interceptDateList = [newDate];
+    } else {
+        interceptDateList.push(newDate);
+    }
+    setInterceptDateList(interceptDateList);
+};
+
+//TODO fix in interation 3 for url-handling module
+formatUrl = function(url) {
+    result = url.split(['//'])[1];
+    result = result.split("").reverse().join("");
+    result = result.split(['/'])[1];
+    result = result.split("").reverse().join("");
+    return result;
 };
 
 /* --------------- ------ Listener functions ------ ---------------*/
 
 replaceListener = function() {
-    chrome.webRequest.onBeforeRequest.removeListener(intercept);
-    addWebRequestListener();
+    chrome.webRequest.onBeforeRequest.removeListener(handleInterception);
+    urlList = filterBlockedSitesOnChecked();
+    if (urlList.length > 0) {
+        addWebRequestListener(urlList);
+    }
+};
+
+filterBlockedSitesOnChecked = function() {
+    if(blockedSites.length > 0) {
+        return blockedSites.filter(function (a) {return a.checkboxVal == true;});
+    }
+    return [];
 };
 
-addWebRequestListener = function() {
-    if(blockedSites != null && blockedSites.length > 0) {
-        chrome.webRequest.onBeforeRequest.addListener(
-                intercept,
-                {
-                    //Url's to be intercepted
-                    urls: blockedSites,
-                    types: ["main_frame"]
-            },
-            ["blocking"]
-        );
+addWebRequestListener = function(urlList) {
+    chrome.webRequest.onBeforeRequest.addListener(
+        handleInterception
+        ,{
+            urls: urlList.map(function (a) {return a.url;})
+            ,types: ["main_frame"]
+        }
+        ,["blocking"]
+    );
+};
+
+intercept = function(details) {
```

Comment:
this function does a bit too many(different) things, because the way it is right now I question whether doing things in this exact order is actually important